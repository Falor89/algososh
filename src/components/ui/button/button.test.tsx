import renderer from 'react-test-renderer'
import { render, screen, fireEvent } from '@testing-library/react'
import { Button } from './button'

describe('Тестирование Button', () => {
    it('Кнопка с текстом рендерится без ошибки', () => {
        const buttonText = renderer
            .create(<Button text='Кнопка с текстом' />)
            .toJSON();
        expect(buttonText).toMatchSnapshot()
    })
    it('Кнопка без текста рендерится без ошибки', () => {
        const buttonWithoutText = renderer
            .create(<Button />)
            .toJSON()
        expect(buttonWithoutText).toMatchSnapshot()
    })
    it('Кнопка с текстом рендерится без ошибки', () => {
        const buttonText = renderer
            .create(<Button disabled />)
            .toJSON();
        expect(buttonText).toMatchSnapshot()
    })
    it('Кнопка без текста рендерится без ошибки', () => {
        const buttonWithoutText = renderer
            .create(<Button isLoader={true} />)
            .toJSON()
        expect(buttonWithoutText).toMatchSnapshot()
    })
    it('Проверяем корректность вызова колбека при клике на кнопку', () => {
        const onClick = jest.fn()
        render(<Button text='test' onClick={onClick} />)
        const button = screen.getByText('test')
        fireEvent.click(button)
        expect(onClick).toHaveBeenCalled()
    })
})