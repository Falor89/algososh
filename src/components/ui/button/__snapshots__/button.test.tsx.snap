// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование Button Кнопка без текста рендерится без ошибки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование Button Кнопка без текста рендерится без ошибки 2`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование Button Кнопка с текстом рендерится без ошибки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка с текстом
  </p>
</button>
`;

exports[`Тестирование Button Кнопка с текстом рендерится без ошибки 2`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
